map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java

```diff
@@ -74,24 +76,29 @@
   @FXML
   private void initialize() {
 
-    // set basemap options
-    basemapList.getItems().addAll(Basemap.createStreets(), Basemap.createImagery(), Basemap
-            .createTopographic(), Basemap.createOceans());
+    // authentication with an API key or named user is required to access basemaps and other location services
+    String yourAPIKey = System.getProperty("apiKey");
+    ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
 
-    // update basemap when selection changes
-    basemapList.getSelectionModel().select(0);
-    basemapList.getSelectionModel().selectedItemProperty()
-            .addListener(o -> map.setBasemap(basemapList.getSelectionModel().getSelectedItem()));
+    // set the basemap style options
+    basemapStyleList.getItems().addAll(BasemapStyle.ARCGIS_STREETS, BasemapStyle.ARCGIS_IMAGERY_STANDARD,
+      BasemapStyle.ARCGIS_TOPOGRAPHIC, BasemapStyle.ARCGIS_OCEANS);
 
-    basemapList.setCellFactory(c -> new BasemapCell());
+    // update the basemap when the selection changes
+    basemapStyleList.getSelectionModel().select(0);
+    basemapStyleList.getSelectionModel().selectedItemProperty()
+      .addListener(o -> map.setBasemap(new Basemap(basemapStyleList.getSelectionModel().getSelectedItem())));
 
-    // create and set a map with the first basemap option
-    map = new ArcGISMap(basemapList.getSelectionModel().getSelectedItem());
+    basemapStyleList.setCellFactory(c -> new BasemapCell());
+    mapView.setMap(map);
+
+    // create a basemap with the first basemap style option and set it to the map
+    Basemap basemap = new Basemap(basemapStyleList.getSelectionModel().getSelectedItem());
+    map = new ArcGISMap(basemap);
     mapView.setMap(map);
 
     // set operational layer options
-    String worldElevationService =
-            "https://sampleserver6.arcgisonline.com/arcgis/rest/services/WorldTimeZones/MapServer";
+    String worldElevationService = "https://sampleserver6.arcgisonline.com/arcgis/rest/services/WorldTimeZones/MapServer";
```

Comment:
I can't remember if there's an effor to not save single-use string variables (especially URLs), and just put them in-line. Meaning you could get rid of this line and just pop the url one line lower?
`ArcGISMapImageLayer worldElevation = new ArcGISMapImageLayer("https://sampleserver6.arcgisonline.com/arcgis/rest/services/WorldTimeZones/MapServer");`
Either way is good though!

Reply:
Yeah we have been doing that I think you're right, I'll do that refactor